netty/src/main/java/io/grpc/transport/netty/Http2Negotiator.java

```diff
@@ -104,26 +111,50 @@ public static ChannelHandler serverTls(SSLEngine sslEngine) {
    * be negotiated, the {@code handler} is added and writes to the {@link Channel} may happen
    * immediately, even before the TLS Handshake is complete.
    */
-  public static ChannelHandler tls(final SSLEngine sslEngine, final ChannelHandler handler) {
-    Preconditions.checkNotNull(sslEngine, "sslEngine");
-    final SettableFuture<Void> completeFuture = SettableFuture.create();
-    if (!installJettyTlsProtocolSelection(sslEngine, completeFuture, false)) {
-      throw new IllegalStateException("NPN/ALPN extensions not installed");
-    }
-    SslHandler sslHandler = new SslHandler(sslEngine, false);
-    sslHandler.handshakeFuture().addListener(
-        new GenericFutureListener<Future<? super Channel>>() {
-          @Override
-          public void operationComplete(Future<? super Channel> future) throws Exception {
-            // If an error occurred during the handshake, throw it to the pipeline.
-            if (future.isSuccess()) {
-              completeFuture.get();
-            } else {
-              future.get();
-            }
+  public static ChannelHandler tls(final SslContext sslContext, final InetSocketAddress inetAddress,
+      final ChannelHandler handler) {
+    Preconditions.checkNotNull(sslContext, "sslContext");
+    Preconditions.checkNotNull(inetAddress, "inetAddress");
+
+    ChannelHandler sslBootstrapHandler = new ChannelHandlerAdapter() {
+      @Override
+      public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+        SSLEngine sslEngine;
+        final SettableFuture<Void> completeFuture = SettableFuture.create();
+        if (isOpenSsl(sslContext.getClass())) {
+          sslEngine = sslContext.newEngine(ctx.alloc());
```

Comment:
I guess that's because it only supports server-side?

We can't use OpenSSL on client-side without that method. That is a deal-breaker, as we won't end up doing certificate verification.